### ISSUE TITLE
BuildRequires does not support boolean operators

### ISSUE THREAD

**Issue by jcsiadal** [rpm-software-management/rpm#1216]
When using the following tag:
BuildRequires: (pkgA or pkgB)
rpmbuild will flag "(PkgA," "pkgB)", and "or" as not found. Removing the parenthesis doesn't resolve the issue. Tested with rpmbuild 4.13 and 4.15.1.

It doesn't seem logical to support the use of boolean operators on installation dependencies at run time, but not to support them at build time. If not supporting this is by design, then rpmlint is failing to flag the use of conditionals in BuildRequires as an error.

I discovered this while building an RPM with a dependency met by either an OS package or the package provided by the product vendor using a different name. Neither RPM "Provides" anything other than their own package names.

**Comment by ignatenkobrain** [comment]
Can you provide a reproducer? We use rich dependencies in BuildRequires at least in 2000 packages in Fedora.

**Comment by jcsiadal** [comment]
Thanks for the fast response. I have to take a mea culpa on this one; I went to rerun the build for exact output and it worked. I spent two hours this morning on two different servers and couldn't get this to work. It works now and I have no idea what changed.

FYI, I went through docs on rpm.org and fedora before posting the issue. Requires, Recommends, and the other dependencies were explicitly listed as supporting Booleans, but BuildRequires was noticeably missing. With the issues I was having, I concluded it may have been left out.

In any case, I'll close this issue. Thanks!
